daffodil-runtime1/src/main/scala/org/apache/daffodil/dpath/NodeInfo.scala

```diff
@@ -643,6 +653,8 @@ object NodeInfo extends Enum {
         case bi: JBigInt => bi.signum >= 0
         case _ => n.longValue >= 0
       }
+
+      override val width: MaybeInt = MaybeInt(32)
```

Comment:
NonNegativeInteger does not have a width. NonNegative integer is the same as a Java BigInteger that can never be negative, so there is no width. This should be MaybeInt.Nope.